zipline/pipeline/loaders/blaze/core.py

```diff
@@ -1104,6 +1104,11 @@ def collect_expr(e, lower):
             materialized_deltas,
             dates,
         )
+        # If we ever have cases where we find out about multiple asof_dates'
+        # data on the same TS, we want to make sure that last_in_date_group
+        # selects the correct last asof_date's value.
+        sparse_output.sort_values(AD_FIELD_NAME, inplace=True)
```

Comment:
Other call sites may need to use different fields to sort the data, and so they would have to tell the function how to check itself anyway.